### ISSUE TITLE
Finetuning

### ISSUE THREAD

**Issue by Abubaker-Nafe** [ultralytics/hub#1094]
### Search before asking

- [x] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and [discussions](https://github.com/ultralytics/hub/discussions) and found no similar questions.

### Question

How can we finetune an already trained model using Ultralytics

### Additional

I have trained a model on a certain dataset, but I want to finetune the model and train it further on another dataset, but I haven't found a way to do so, since every time I train the already trained model, it looks like it trains it all over again without taking the first one's progress.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Abubaker-Nafe, thank you for raising an issue about [Ultralytics HUB](https://hub.ultralytics.com) 🚀! Please visit our [HUB Docs](https://docs.ultralytics.com/hub) to learn more:

- [**Quickstart**](https://docs.ultralytics.com/hub/quickstart). Start training and deploying YOLO models with HUB in seconds.
- [**Datasets: Preparing and Uploading**](https://docs.ultralytics.com/hub/datasets). Learn how to prepare and upload your datasets to HUB in YOLO format.
- [**Projects: Creating and Managing**](https://docs.ultralytics.com/hub/projects). Group your models into projects for improved organization.
- [**Models: Training and Exporting**](https://docs.ultralytics.com/hub/models). Train YOLOv5 and YOLOv8 models on your custom datasets and export them to various formats for deployment.
- [**Integrations**](https://docs.ultralytics.com/hub/integrations). Explore different integration options for your trained models, such as TensorFlow, ONNX, OpenVINO, CoreML, and PaddlePaddle.
- [**Ultralytics HUB App**](https://docs.ultralytics.com/hub/app). Learn about the Ultralytics App for iOS and Android, which allows you to run models directly on your mobile device.
    * [**iOS**](https://docs.ultralytics.com/hub/app/ios). Learn about YOLO CoreML models accelerated on Apple's Neural Engine on iPhones and iPads.
    * [**Android**](https://docs.ultralytics.com/hub/app/android). Explore TFLite acceleration on mobile devices.
- [**Inference API**](https://docs.ultralytics.com/hub/inference-api). Understand how to use the Inference API for running your trained models in the cloud to generate predictions.

If this is a 🐛 Bug Report, please provide screenshots and steps to [reproduce](https://docs.ultralytics.com/help/minimum_reproducible_example/) your problem to help us get started working on a fix.

If this is a ❓ Question, please provide as much information as possible, including dataset, model, and environment details so that we might provide the most helpful response.

This is an automated response 🤖. An Ultralytics engineer will review your issue and assist you soon. Thank you for your patience!

**Comment by pderrenger** [comment]
Hi @Abubaker-Nafe! 👋

Great question about fine-tuning previously trained models in Ultralytics HUB. Fine-tuning is indeed a powerful technique that lets you leverage knowledge from an existing model and adapt it to new data without starting from scratch.

## How to Fine-tune Models in Ultralytics HUB

When training a new model in Ultralytics HUB, you can use a previously trained model as your starting point by following these steps:

1. Navigate to the [Models](https://hub.ultralytics.com/models) page and click on **Train Model**
2. Complete the Dataset selection step
3. In the Model step, click on the **Custom** tab
4. Select your previously trained model from the list of your existing models
5. Configure any additional training parameters as needed
6. Continue to the Train step and start training using your preferred method (Cloud Training, Google Colab, or your own agent)

This approach ensures that instead of starting from scratch, the training process will build upon the weights and learning from your previous model.

## Technical Tips for Effective Fine-tuning

To ensure fine-tuning works effectively:

- Use a lower learning rate than the original training (you can adjust this in the Advanced Model Configuration settings)
- Consider using a shorter training duration since you're building on existing knowledge
- Make sure your new dataset is properly formatted and aligned with the classes/labels from your original training

If you're using the Python API with your own agent for training, you can also use this approach:

```python
from ultralytics import YOLO

# Load your previously trained model
model = YOLO('path/to/your/previously_trained_model.pt')

# Fine-tune the model on new data
results = model.train(data='path/to/new/dataset.yaml', epochs=10)
```

## Common Fine-tuning Issues

If you're still experiencing the model training "from scratch," check that:

1. You're actually selecting your trained model (not a fresh pre-trained model)
2. Your training parameters aren't set to overwrite previous learning (particularly learning rate and epochs)
3. The new dataset is compatible with the original model architecture

For more detailed information about fine-tuning, check out our [glossary entry on fine-tuning](https://www.ultralytics.com/glossary/fine-tuning), which explains the concept in depth.

Hope this helps! Let me know if you have any further questions about the fine-tuning process in Ultralytics HUB.
